service/discovery/aws/compute.go

```diff
@@ -0,0 +1,243 @@
+		resources = append(resources, &function)
+	}
+
+	return
+}
+
+// discoverVirtualMachines discovers all VMs (in the current region)
+func (d *computeDiscovery) discoverVirtualMachines() ([]voc.VirtualMachine, error) {
+	resp, err := d.virtualMachineAPI.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{})
+	if err != nil {
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			err = formatError(ae)
+		}
+		return nil, err
+	}
+	var resources []voc.VirtualMachine
+	for _, reservation := range resp.Reservations {
+		for i := range reservation.Instances {
+			vm := &reservation.Instances[i]
+			computeResource := &voc.Compute{
+				CloudResource: &voc.CloudResource{
+					ID:   d.getARNOfVM(vm),
+					Name: d.getNameOfVM(vm),
+					// TODO(all): -1 or 0 as default value when no creation time is available?
+					CreationTime: -1,
+					Type:         []string{"VirtualMachine", "Compute", "Resource"},
+				},
+			}
+
+			resources = append(resources, voc.VirtualMachine{
+				Compute:          computeResource,
+				NetworkInterface: d.getNetworkInterfacesOfVM(vm),
+				BlockStorage:     d.getBlockStorageIDsOfVM(vm),
+				Log:              d.getLogsOfVM(vm),
+			})
+		}
+	}
+	return resources, nil
+}
+
+// discoverFunctions discovers all lambda functions
+// TODO(all): FunctionVersion in input to ALL ok? (=all versions of a lambda functions are discovered)
+// TODO(oxisto): Is this a good approach with "for hasNextMarker" loop (I would use overloaded methods - not in Go)
+func (d *computeDiscovery) discoverFunctions() ([]voc.Function, error) {
+	// 'listFunctions' discovers up to 50 Lambda functions per execution
+	resp, err := d.functionAPI.ListFunctions(context.TODO(), &lambda.ListFunctionsInput{
+		FunctionVersion: types2.FunctionVersionAll,
+	})
+
+	if err != nil {
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			err = formatError(ae)
+		}
+		return nil, err
+	}
+
+	var resources []voc.Function
+	resources = append(resources, d.getFunctionResources(resp.Functions)...)
+
+	// Execute 'listFunctions' in a loop until all Lambda functions are discovered
+	// If nextMarker in resp is set, more functions are discovered (setting this marker in the input parameter)
+	hasNextMarker := resp.NextMarker != nil
+	for hasNextMarker {
+		resp, err = d.functionAPI.ListFunctions(context.TODO(), &lambda.ListFunctionsInput{
+			Marker:          resp.NextMarker,
+			FunctionVersion: types2.FunctionVersionAll,
+		})
+		if err != nil {
+			var ae smithy.APIError
+			if errors.As(err, &ae) {
+				err = formatError(ae)
+			}
+			return nil, err
+		}
+		resources = append(resources, d.getFunctionResources(resp.Functions)...)
+		hasNextMarker = resp.NextMarker != nil
+	}
+
+	return resources, nil
+}
+
+// getFunctionResources iterates functionConfigurations and returns a list of corresponding FunctionResources
+func (d *computeDiscovery) getFunctionResources(functions []types2.FunctionConfiguration) (resources []voc.Function) {
```

Comment:
`getFunctionResources` does not sound like good name for this. Maybe something more like `map` or `translate`?

Reply:
Yes, `map` is better. Also will change similar methods.